lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,251 @@
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item * C<$allowed_naked_packages> - hashref of number of allowed naked subs, keyed by package name (optional)
+
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+=back
+
+=cut
+
+sub check_pod_coverage {
+    my $path                   = shift;
+    my $allowed_naked_packages = shift;
+    my $ignored_packages       = shift;
+
+    check_allowed_naked_packages($allowed_naked_packages, $ignored_packages) if keys %$allowed_naked_packages;
+
+    # Check for newly added packages PODs
+    my @ignored_packages = (keys %$allowed_naked_packages, @$ignored_packages);
+    foreach my $package (Test::Pod::Coverage::all_modules(@$path)) {
+        next if @ignored_packages && (any { $_ eq $package } @ignored_packages);
+        pod_coverage_ok($package, {private => []});
+    }
+}
+
+=head2 check_pod_syntax
+
+Check POD syntax for all the modules that exists under a given directory.
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+=back
+
+=cut
+
+sub check_pod_syntax {
+    my $path             = shift;
+    my $ignored_packages = shift;
+    my $Test_Builder     = Test::More->builder;
+
+    my @ignored_packages_full_path = ();
+    for (@$ignored_packages) {
+        my $file_path = module_path($_);
+        push @ignored_packages_full_path, $file_path if defined $file_path;
+    }
+
+    my @files_path = File::Find::Rule->file()->name('*.p[m|l]')->in(@$path);
+
+    for my $file_path (@files_path) {
+        chomp $file_path;
+        next if any { /\Q$file_path/ } @ignored_packages_full_path;
+
+        my $check_result = podchecker($file_path);
+        if ($check_result == POD_SYNTAX_IS_OK) {
+            $Test_Builder->ok(1, sprintf("Pod structure is OK in the file %s.", $file_path));
+        } elsif ($check_result == FILE_HAS_NO_POD) {
+            $Test_Builder->todo_skip(sprintf("There is no POD in the file %s.", $file_path));
+        } else {
+            $Test_Builder->ok(0, sprintf("There are %d errors in the POD structure in the %s.", $check_result, $file_path));
+        }
+    }
+}
+
+=head2 check_allowed_naked_packages
+
+Checks passed allowed_naked_packages against existing package files.
+
+=over 4
+
+=item * C<$allowed_naked_packages> - hashref of number of allowed naked subs, keyed by package name (optional)
+
+=item * C<$ignored_packages> - a list of packages that will be ignored in our checks, supports arrayref (optional)
+
+=back
+
+Possible results
+
+=over 4
+
+=item - B<todo fail> if the numbers of existing naked subs are equal to passed value.
+
+=item - B<fails> if the numbers of existing naked subs are more/less than the passed value.
+
+=item - B<fails> if a package has 100% POD coverage and it passed as a L<$allowed_naked_package>.
+
+=back
```

Comment:
Is it OK to show possible results like this?

Reply:
yes but again your lists are inconsistent - `*` for the bullet point, don't use `-` since you *already* use `*` earlier...

Reply:
Sure, I just wanted to make a difference between arguments and results. I'll replace them with *